Unit tests for 0.8.4 fail on macOS

I'm trying to fix PR #306. However, when I try to `cargo test` on macOS 10.14.4 / rustc 1.36.0-nightly, `cargo test` gives these failures:

```
failures:
    test_alias
    test_annotation
    test_asserted_cast
    test_derive_eq
    test_display_list
    test_enum
    test_item_types
    test_item_types_renamed
    test_monomorph_1
    test_monomorph_3
    test_must_use
    test_prefix
    test_rename
    test_reserved
    test_transform_op

test result: FAILED. 54 passed; 15 failed; 0 ignored; 0 measured; 0 filtered out
```

There are many pages of error output, but here's an example:

```
thread 'test_reserved' panicked at 'Output failed to compile: Output { status: ExitStatus(ExitStatus(256)), stdout: "", stderr: "/Users/rb/Documents/Development/Source/_thirdparty/cbindgen/tests/expectations/reserved.cpp:16:8: warning: scoped enumerations are a C++11 extension [-Wc++11-extensions]\n  enum class Tag : uint8_t {\n       ^\n/Users/rb/Documents/Development/Source/_thirdparty/cbindgen/tests/expectations/reserved.cpp:16:14: error: ISO C++ forbids forward references to \'enum\' types\n  enum class Tag : uint8_t {\n             ^\n/Users/rb/Documents/Development/Source/_thirdparty/cbindgen/tests/expectations/reserved.cpp:16:28: error: expected \'(\' for function-style cast or type construction\n  enum class Tag : uint8_t {\n                   ~~~~~~~ ^\n1 warning and 2 errors generated.\n" }', tests/tests.rs:77:5
```

--- a/tests/tests.rs
+++ b/tests/tests.rs
@@ -71,6 +71,10 @@ fn compile(cbindgen_output: &Path, language: Language) {
     command.arg("-D").arg("DEFINED");
     command.arg("-c").arg(cbindgen_output);
     command.arg("-o").arg(&object);
+    if let Language::Cxx = language {
+        // enum class is a c++11 extension which makes g++ on macos 10.14 error out
+        command.arg("-std=c++11");
+    }
 
     println!("Running: {:?}", command);
     let out = command.output().expect("failed to compile");

